extmod/modssl_mbedtls.c

```diff
@@ -138,6 +156,41 @@ STATIC NORETURN void mbedtls_raise_error(int err) {
     #endif
 }
 
+#ifdef MICROPY_SSL_MBEDTLS_EXTRAS
+STATIC void ssl_handle_handshake_error(mp_obj_ssl_socket_t *sslsock, int *errcode) {
+
+    uint8_t alert = MBEDTLS_SSL_ALERT_MSG_HANDSHAKE_FAILURE;
+    uint32_t flags = 0;
+    uint ret = 0;
+    if (*errcode == MBEDTLS_ERR_X509_CERT_VERIFY_FAILED) {
+
+        // The certificate may have been rejected for several reasons.
+        flags = mbedtls_ssl_get_verify_result(&sslsock->ssl);
+
+    } else if (*errcode == MBEDTLS_ERR_SSL_NO_CLIENT_CERTIFICATE) {
+        /* alert = MBEDTLS_SSL_ALERT_MSG_CERT_REQUIRED; tlsv1.3 */
+
+        mbedtls_ssl_send_alert_message(&sslsock->ssl, MBEDTLS_SSL_ALERT_LEVEL_FATAL,
+            alert);
+    }
+
+    sslsock->sock = MP_OBJ_NULL;
+    mbedtls_ssl_free(&sslsock->ssl);
+
+    if (*errcode == MBEDTLS_ERR_X509_CERT_VERIFY_FAILED) {
+        char xcbuf[512];
+        ret = mbedtls_x509_crt_verify_info(xcbuf, sizeof(xcbuf), "\n", flags);
+        // The length of the string written (not including the terminated nul byte),
+        // or a negative err code.
+        if (ret > 0) {
+            mp_raise_ValueError(MP_ERROR_TEXT(xcbuf));
```

Comment:
This needs to be changed to something like:

`mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("%s"), xcbuf);`

Otherwise this PR works great!

Reply:
Yes, I did notice that the error message wasn't properly handled and I guess garbage collected? But I didn't know how to fix it and I haven't looked into it yet, so thanks @peterzuger for testing this PR and providing the fix 🥇 , I will add it asap 👍🏼